src/main/java/xyz/nifeather/morph/network/multiInstance/slave/NetworkDataHolder.java

```diff
@@ -50,106 +48,152 @@ public NetworkDataHolder(SlaveInstance bindingSlave)
     }
 
     @Override
-    public List<DisguiseMeta> getAvailableDisguisesFor(Player player)
+    public CompletableFuture<PlayerMeta> loadAsync(UUID uuid)
     {
-        var playerMeta = getPlayerMeta(player);
+        var future = new CompletableFuture<PlayerMeta>();
+
+        bindingSlave.requestData(uuid).thenAccept(u ->
+        {
+            var data = nullablePlayerMeta(u);
 
-        return playerMeta.getUnlockedDisguises();
+            if (data != null)
+                future.complete(data);
+            else
+                future.completeExceptionally(new NullDependencyException("The future of the requested data has been finished, but we can't find a matching data in NetworkDataHolder!"));
+        });
+
```

Comment:
This is kind of tricky, and I'm not sure if it's fine. Maybe requesting data should now be `NetworkDataHolder`'s job?